sdk/src/asset_handlers/woff_io.rs

```diff
@@ -0,0 +1,2050 @@
+        write!(f, "{}{}{}{}", self.data[0], self.data[1], self.data[2], self.data[3])
+    }
+}
+
+impl std::fmt::Debug for TableTag {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}{}{}{}", self.data[0], self.data[1], self.data[2], self.data[3])
+    }
+}
+
+/// Ad-hoc "tag" names for font-file entities which are not really tables,
+/// but which it is frequently convenient to treat as if they were. We re-use
+/// the font-file magic strings, because they are extremely unlikely to appear
+/// as table tags.
+const WOFF_HEADER_TAG: TableTag = TableTag { data: *b"wOFF", };
+const WOFF_METADATA_TAG: TableTag = TableTag { data: *b"wXML", };
+const WOFF_PRIVATE_DATA_TAG: TableTag = TableTag { data: *b"wPRV", };
+/// Pseudo-tag for the table directory.
+const TABLE_DIRECTORY_TAG: TableTag = TableTag { data: *b"\t\t\t\t", };
+/// Tag for the 'C2PA' table.
+const C2PA_TABLE_TAG: TableTag = TableTag { data: *b"C2PA", };
+/// Tag for the 'head' table in a font.
+const HEAD_TABLE_TAG: TableTag = TableTag { data: *b"head", };
+
+/// 32-bit font-format identifier.
+///
+/// Embedded OpenType and MicroType Express formats cannot be detected with a
+/// simple magic-number sniff. Conceivably, EOT could be dealt with as a
+/// variation on SFNT, but MTX needs more exotic handling.
+enum Magic {
+    /// 'OTTO' - OpenType
+    OpenType = 0x4f54544f,
+    /// FIXED 1.0 - TrueType (or possibly v1.0 Embedded OpenType)
+    TrueType = 0x00010000,
+    /// 'wOFF' - WOFF 1.0
+    Woff = 0x774f4646,
+    /// 'wOF2' - WOFF 2.0
+    Woff2 = 0x774f4632,
+}
+
+/// Used to attempt conversion from u32 to a Magic value.
+impl TryFrom<u32> for Magic {
+    type Error = ();
+
+    /// Tries to convert from u32 to a valid font version.
+    fn try_from(v: u32) -> core::result::Result<Self, Self::Error> {
+        match v {
+            x if x == Magic::TrueType as u32 => Ok(Magic::TrueType),
+            x if x == Magic::OpenType as u32 => Ok(Magic::OpenType),
+            x if x == Magic::Woff as u32 => Ok(Magic::Woff),
+            x if x == Magic::Woff2 as u32 => Ok(Magic::Woff2),
+            _ => Err(()),
+        }
+    }
+}
+
+/// 'C2PA' font table - in storage
+#[derive(Debug)]
+#[repr(C, packed)]       // As defined by the C2PA spec.
+#[allow(non_snake_case)] // As named by the C2PA spec.
+struct TableC2PARaw {
+    majorVersion: u16,
+    minorVersion: u16,
+    activeManifestUriOffset: u32,
+    activeManifestUriLength: u16,
+    manifestStoreOffset: u32,
+    manifestStoreLength: u32,
+}
+impl TableC2PARaw {
+    fn write<TDest: Write + ?Sized>(&mut self, destination: &mut TDest) -> Result<()> {
+        destination.write_u16::<BigEndian>(self.majorVersion)?;
+        destination.write_u16::<BigEndian>(self.minorVersion)?;
+        destination.write_u32::<BigEndian>(self.activeManifestUriOffset)?;
+        destination.write_u16::<BigEndian>(self.activeManifestUriLength as u16)?;
```

Comment:
An unnecessary cast!

Reply:
Weird that he doesn't complain about it. Reminds me - I wanted to point out again that the fourth field is misaligned -- maybe we should add a reserved `u16` before/after this field, to get back on a 4-byte boundary.

Reply:
It does complain if you run `cargo clippy --all-features --all-targets -- -D warnings` for example. And since we are using u16 and u32, I think we do have 4-byte alignment.